src/core/grant-authorization.ts

```diff
@@ -15,10 +15,16 @@ export class GrantAuthorization {
   public static async authorizeGenericMessage(
     tenant: string,
     incomingMessage: Message<GenericMessage>,
-    author: string,
     messageStore: MessageStore,
-  ): Promise<void> {
+  ): Promise<PermissionsGrantMessage> {
     const permissionsGrantId: string = incomingMessage.authorizationPayload!.permissionsGrantId!;
+    if (incomingMessage.author === undefined) {
```

Comment:
Sanity: I "think" this code path should not be reachable externally. Was there a particular code path that prompted this check? No push back if this is just a  defensive programming.